src/main/java/Duke.java

```diff
@@ -1,10 +1,151 @@
+        System.out.println("Hope it helps!! woof a nice day ੯•໒꒱❤︎");
+        printLine();
+    }
+    public static void startProgram() {
+        Scanner scan = new Scanner(System.in);
+        String s;
+        s = scan.nextLine();
+        while (s.trim().isEmpty() || s.trim().charAt(0) == '#') {
+            s = scan.nextLine();
+        }
+        final String[] split = s.trim().split("\\s+", 2);
+        final String[] commandTypeAndParams = split.length == 2 ? split : new String[]{split[0], ""};
+        final String commandType = commandTypeAndParams[0];
+        final String commandArgs = commandTypeAndParams[1];
+        switch (commandType) {
+        case "bye":
+            goodBye();
+            return;
+        case "list":
+            printListOfTasks();
+            break;
+        case "todo":
+            addTodo(commandArgs);
+            break;
+        case "deadline":
+            addDeadline(commandArgs);
+            break;
+        case "event":
+            addEvent(commandArgs);
+            break;
+        case "mark":
+            markTask(commandArgs);
+            break;
+        case "unmark":
+            unmarkTask(commandArgs);
+            break;
+        default:
+            info();
+        }
+
+        startProgram();
+    }
+
+    private static void unmarkTask(String commandArgs) {
+        final int unmarkId = Integer.parseInt(commandArgs)-1;
+        if (!tasks[unmarkId].isDone) {
+            System.out.println("This task hasn't been marked as done yet ∪･ω･∪");
+        } else {
+            tasks[unmarkId].markAsNotDone();
+            System.out.println("I've unmarked this task ∪･ω･∪:");
+            System.out.println(tasks[unmarkId]);
+        }
+        printLine();
+    }
+
+    private static void markTask(String commandArgs) {
+        final int markId = Integer.parseInt(commandArgs)-1;
+        if (tasks[markId].isDone) {
+            System.out.println("This task has already been marked as done ੯•໒꒱❤︎");
+        } else {
+            tasks[markId].markAsDone();
+            System.out.println("I've marked this task as done ੯•໒꒱❤︎:");
+            System.out.println(tasks[markId]);
+        }
+        printLine();
+    }
+
+    private static void addEvent(String commandArgs) {
+        final int indexOfFrom = commandArgs.indexOf("/from");
+        final int indexOfTo = commandArgs.indexOf("/to");
+        String eventDescription = commandArgs.substring(0,indexOfFrom).trim();
+        String from = commandArgs.substring(indexOfFrom, indexOfTo).trim().replace("/from", "");
+        String to = commandArgs.substring(indexOfTo, commandArgs.length()).trim().replace("/to", "");
+        addTask(new Event(eventDescription, from, to));
+        printLine();
+        System.out.println("Got it. I've added this task: \n" + tasks[taskCount -1]);
+        System.out.println("Now you have " + taskCount + " tasks in your list.");
+        printLine();
```

Comment:
Perhaps this 4 lines can be made into a new method, to improve readability? It seems repeated in many other methods and it may help in structuring the code logically.